### PR TITLE
SX: add support for runtime styles

### PR DESCRIPTION
**Pull request by mrtnzlml**
Runtime styles should be now working. You can test it by running Relay Example or SX Tailwind Website and try to go from page to page. You should see that the styles work as expected without the need to SSR the styles.

Closes: https://github.com/adeira/universe/issues/1012

TODO:

- [x] merge https://github.com/adeira/universe/pull/1239
- [x] revert https://github.com/adeira/universe/pull/1244

### CODE REVIEW

File: src/example-relay/src/pages/_document.js [hunk 1]
```diff
@@ -34,6 +32,7 @@ export default class MyDocument extends Document {
             href="https://fonts.googleapis.com/css?family=Roboto:400,400i,500,500i,700"
             rel="stylesheet"
           />
+          <style dangerouslySetInnerHTML={{ __html: this.props.css }} />
```
Comment (tbergq) [inline src/example-relay/src/pages/_document.js:35]: won't this add your style tag 2 times?

File: src/sx/src/create.js [hunk 2]
```diff
@@ -127,6 +130,23 @@ export default function create<T: SheetDefinitions>(
         objectType.startsWith(':') ? objectType : '',
       );
 
+      if (typeof window !== 'undefined') {
```
Comment (tbergq) [inline src/sx/src/create.js:133]: `isBrowser` from `adeira/js`?

File: src/example-relay/src/pages/_document.js (hunk 1 above)
Comment (mrtnzlml) [reply src/example-relay/src/pages/_document.js:35]: Yes, it should not be there - rebase mistake.

File: src/sx/src/create.js [hunk 3]
```diff
@@ -50,7 +51,14 @@ export default function create<T: SheetDefinitions>(
     `Function 'sx.create' cannot be called with empty stylesheet definition.`,
   );
 
-  const { hashRegistry } = styleCollector.collect(sheetDefinitions);
+  const { hashRegistry, styleBuffer } = styleCollector.collect(sheetDefinitions);
+
+  if (
+    isBrowser() &&
+    process.env.NODE_ENV !== 'test' // TODO: how to do it better (?)
```
Comment (tbergq) [inline src/sx/src/create.js:58]: What's the issue with the test environment?

File: src/sx/src/injectRuntimeStyles.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+opaque type StyleElementType = HTMLStyleElement | null;
+let styleAdeiraSXTag: StyleElementType = null;
+
+// https://developer.mozilla.org/en-US/docs/Web/API/CSS_Object_Model
+export default function injectRuntimeStyles(styleBuffer: StyleBufferType) {
+  if (styleAdeiraSXTag === null) {
+    styleAdeiraSXTag = ((document.querySelector('style[data-adeira-sx]'): any): StyleElementType);
+    invariant(
+      styleAdeiraSXTag != null,
+      'SX cannot render any styles because <style data-adeira-sx /> was not found in the document.',
```
Comment (tbergq) [inline src/sx/src/injectRuntimeStyles.js:22]: Alternatively, we could create it, no?

**Comment by tbergq** [comment]
Tested it in relay-example and it seems to be working fine 😊 Good job 👍

**Review by tbergq — APPROVED** [review]

File: src/sx/src/create.js (hunk 3 above)
Comment (mrtnzlml) [reply src/sx/src/create.js:58]: It wants the element to be present somewhere in the document (see the invariant you commented on below).

Comment (tbergq) [reply src/sx/src/create.js:58]: Right, so creating it if it does not exists would solve this as well.

Comment (mrtnzlml) [reply src/sx/src/create.js:58]: Most likely yes. I will look at it if I should do it now or in the next PR.

**Comment by mrtnzlml** [comment]
@tbergq Please have a look again. I added the automatic `style` tag creation.

File: src/sx/src/injectRuntimeStyles.js (hunk 4 above)
Comment (mrtnzlml) [reply src/sx/src/injectRuntimeStyles.js]: yup, done

**Review by tbergq — APPROVED** [review]

**Pull request merged**